src/compiler/program.ts

```diff
@@ -1463,6 +1477,25 @@ namespace ts {
                 }
             });
 
+            if (!options.disableSizeLimit && file && file.text && !hasTypeScriptFileExtension(file.fileName)) {

```

Comment:
looks like we do not need this in the program building, since we are doing it in the server.